docs/contributing.rst

```diff
@@ -161,37 +171,75 @@ you can delete it by doing ``git branch -D simulation-enhancements``.
 Obviously this isn't a complete guide to git, but hopefully it
 jump-starts the git learning process.
 
+Testing
+=======
+
+SNCosmo uses pytest to check that all of the code is running as expected. When
+you add new functionality to SNCosmo, you should write a test for that
+functionality. All of the tests can be found in the ``sncosmo/tests``
+directory.
+
+When a new PR is created, the testsuite will be run automatically on a range
+of different machines and conditions using `tox`. You can run these same tests
+locally using ``tox``. First, install `tox`::
+
+      pip install tox
+
+From within the SNCosmo directory, run the test suite::
+
+      tox -e py3
+
+The previous command will run the core test suite with the currently installed
+version of Python. You can run the full test suite with all of the optional
+dependencies by adding the ``-alldeps`` tag::
+
+      tox -e py3-alldeps
+
+Running the tests with the ``-cov`` tag will generate a coverage report::
+
+      tox -e py3-cov
+
+``tox`` can also be used to check the code style::
+
+      tox -e codestyle
+
+or to build the documentation::
+
+      tox -e build_docs
+
+``tox`` uses virtual environments for testing which can be somewhat slow. You
+can alternatively run the test in your own Python environment. First, install
+all of the testing dependencies from the ``test`` section of ``setup.cfg``.
+This can be done automatically when installing SNCosmo with the following
+command::
+
+      pip install -e .[test]
+
+The tests can then be run with the following command::
+
+      pytest --pyargs sncosmo
+
 
 Developer's documentation: release procedure
 ============================================
 
-These are notes mainly for the one person that manages releases.
-Yes, this could be more automated, but it isn't done very often,
-and involves some human verification.
+The release procedure is automated through GitHub Actions. To create a new
+release:
 
 - Update ``docs/history.rst`` with a summary of the new version's changes.
-- Bump version in ``sncosmo/__init__.py``.
-- Build package and docs and check that docs look good.
-- Commit.
-- ``git clean -dfx``
-- ``tox`` (test in clean virtual env and make sure everything passes.
-- ``setup.py sdist upload`` (register new version and upload.)
-
-**Post-release steps:**
```

Comment:
Also, do we not need any new post-release prep for the next version?

Reply:
Nope, everything is automated now. As soon as you publish a new release on GitHub, everything else should happen automatically.

Reply:
I guess that the one step that isn't automated is creating the feature branch (e.g. 2.5.x). We have never really done anything with those though, so I'm not convinced that they are necessary. I'd prefer to just stop making them.